### PR TITLE
Refs [Task] [34400] Load gifs in Home Screen

### PR DESCRIPTION
**Pull request by vvanan158**
## Related Tickets
- [#Task 34400: Load gifs in Home Screen](https://edu-redmine.sun-asterisk.vn/issues/34400)
## WHAT
- Show list gifs trending in home screen
## Evidence (Screenshot or Video)
image
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)

### CODE REVIEW

File: app/src/main/java/com/sun/imagegif/ui/home/HomeFragment.kt [hunk 1]
```diff
@@ -17,6 +29,41 @@ class HomeFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_home, container, false)
     }
 
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        initViews()
+        getString(R.string.search)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/imagegif/ui/home/HomeFragment.kt]: đoạn code này dùng làm gì e?

File: app/src/main/java/com/sun/imagegif/ui/home/HomeFragment.kt [hunk 2]
```diff
@@ -17,6 +29,41 @@ class HomeFragment : Fragment() {
...
+    }
+
+    override fun onStart() {
+        presenter.run {
+            setView(this@HomeFragment)
+            getTrending()
+        }
+        super.onStart()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/imagegif/ui/home/HomeFragment.kt]: sao thằng super lại nằm dưới vậy e?

File: app/src/main/java/com/sun/imagegif/ui/home/adapter/TrendingAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.imagegif.data.model.Gif
+import com.sun.imagegif.utils.OnItemRecyclerViewClickListener
+import com.sun.imagegif.utils.OnItemViewHolderClickListener
+
+class TrendingAdapter : RecyclerView.Adapter<TrendingViewHolder>(),
+    OnItemViewHolderClickListener<Gif> {
+
+    private val items: MutableList<Gif> = mutableListOf()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/imagegif/ui/home/adapter/TrendingAdapter.kt]:
```suggestion
    private val items = mutableListOf<Gif>()
```

File: app/src/main/res/layout/item_gif_trending.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="@dimen/dp_180"
+    android:layout_height="@dimen/dp_220"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/item_gif_trending.xml]: chơi set cứng view luôn vậy e?

File: app/src/main/java/com/sun/imagegif/ui/home/HomeFragment.kt (hunk 1 above)
Comment (vvanan158) [reply app/src/main/java/com/sun/imagegif/ui/home/HomeFragment.kt]: í e quên xóa

File: app/src/main/res/layout/item_gif_trending.xml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent">
+
+    <androidx.cardview.widget.CardView
+        android:layout_width="0dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/item_gif_trending.xml:8]: chuyển card view thành parent view luôn đc không chú

Comment (vvanan158) [reply app/src/main/res/layout/item_gif_trending.xml:8]: không được a ạ

Comment (vvanan158) [reply app/src/main/res/layout/item_gif_trending.xml:8]: không được a ạ

Comment (vvanan158) [reply app/src/main/res/layout/item_gif_trending.xml:8]: đặt cardview bên trong ConstraintLayout mới dùng đc  a ạ
```
app:layout_constraintDimensionRatio="180:220" 
```

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**